### ISSUE TITLE
Logs are very hard to parse

### ISSUE THREAD

**Issue by petemounce** [aws/aws-sdk-net#157]
I'm trying to parse the logs that the SDK emits when 
```
    <add key="AWSLogging" value="SystemDiagnostics"/>
    <add key="AWSLogMetrics" value="true"/>
    <add key="ResponseLogging" value="OnError"/>
```

I get logs that look like 
```
Request metrics: AsyncCall = False; CanonicalRequest = POST
/123456789012/uk-dev-someapi-somequeue

content-type:application/x-www-form-urlencoded; charset=utf-8
host:sqs.eu-west-1.amazonaws.com
user-agent:aws-sdk-dotnet-45/2.3.5.0 .NET Runtime/4.0 .NET Framework/4.0 OS/6.2.9200.0 ClientSync
x-amz-content-sha256:b214f949fe6fc0f0d7e732a51779aa2e3498e683046c6a82272c6626f3fe93f1
x-amz-date:20141230T210136Z

content-type;host;user-agent;x-amz-content-sha256;x-amz-date
b214f949fe6fc0f0d7e732a51779aa2e3498e683046c6a82272c6626f3fe93f1; StringToSign = AWS4-HMAC-SHA256
20141230T210136Z
20141230/eu-west-1/sqs/aws4_request
640ef125aa0f927697be5778b11d743030f6ee28cfcea03cc0510ffe14abe3d8; ServiceName = Amazon.SQS; ServiceEndpoint = https://sqs.eu-west-1.amazonaws.com/; MethodName = ReceiveMessageRequest; RequestSize = 82; StatusCode = OK; BytesProcessed = 240; AWSRequestID = 83dbebf9-3646-5921-858e-a6e422a9f238; CredentialsRequestTime = 00:00:00.0000025; RequestSigningTime = 00:00:00.0002638; HttpRequestTime = 00:00:20.0520366; ResponseUnmarshallTime = 00:00:00.0000903; ResponseProcessingTime = 00:00:00.0001057; ClientExecuteTime = 00:00:20.0526334; 
```

I would find these easier to parse if

0. the log was either 
  * structured data (eg CSV, JSON, key/value pair CSV, XML (I suppose))
  * had a schema or formal spec
0. the field separator was not a newline (since newlines can apparently be part of field values legitimately)
0. TimeSpans were serialised to microseconds (since you have precision beneath milliseconds), not `TimeSpan.ToString()`
  * and the unit given in the log - eg `403043us` (please don't use an actual greek letter `mu` ;-) )

The challenge is that I need to create a regex to parse the whole thing, which will obviously be very brittle. The absence of a formal spec suggests that the log output may change silently between SDK versions (I didn't find tests that cover the log/diagnostics features, but I also only looked via the GitHub file finder, not in-depth).

It would also be nice if you included the numerical http status code of the response, instead of / as well as the name, since that's more flexible (I can derive the name from it more easily, and I can do arithmetic on it directly instead of looking up first).

**Comment by jimfl** [comment]
Thanks for bringing this to our attention! We liked this idea so much that we added it in 2.3.16. You can configure for JSON output, or if that doesn't meet your needs you can configure a custom formatter.

You can read more about how to set it up in this [blog post][1].

[1]: http://aws-blogs.integ.amazon.com/net/post/TxZBAE2YX8FMQ5/Alternative-Formatting-for-Metrics-Data-in-NET-SDK-Logs
